bindings/c/manifoldc.cpp

```diff
@@ -0,0 +1,611 @@
+  auto second = new (mem_second) Manifold(pair.second);
+  return {to_c(first), to_c(second)};
+}
+
+ManifoldManifoldPair manifold_split_by_plane(void *mem_first, void *mem_second,
+                                             ManifoldManifold *m,
+                                             float normal_x, float normal_y,
+                                             float normal_z, float offset) {
+  auto normal = glm::vec3(normal_x, normal_y, normal_z);
+  auto pair = from_c(m)->SplitByPlane(normal, offset);
+  auto first = new (mem_first) Manifold(pair.first);
+  auto second = new (mem_second) Manifold(pair.second);
+  return {to_c(first), to_c(second)};
+}
+
+ManifoldManifold *manifold_trim_by_plane(void *mem, ManifoldManifold *m,
+                                         float normal_x, float normal_y,
+                                         float normal_z, float offset) {
+  auto normal = glm::vec3(normal_x, normal_y, normal_z);
+  auto trimmed = from_c(m)->TrimByPlane(normal, offset);
+  return to_c(new (mem) Manifold(trimmed));
+}
+
+ManifoldManifold *manifold_translate(void *mem, ManifoldManifold *m, float x,
+                                     float y, float z) {
+  auto v = glm::vec3(x, y, z);
+  auto translated = from_c(m)->Translate(v);
+  return to_c(new (mem) Manifold(translated));
+}
+
+ManifoldManifold *manifold_rotate(void *mem, ManifoldManifold *m, float x,
+                                  float y, float z) {
+  auto rotated = from_c(m)->Rotate(x, y, z);
+  return to_c(new (mem) Manifold(rotated));
+}
+
+ManifoldManifold *manifold_scale(void *mem, ManifoldManifold *m, float x,
+                                 float y, float z) {
+  auto s = glm::vec3(x, y, z);
+  auto scaled = from_c(m)->Scale(s);
+  return to_c(new (mem) Manifold(scaled));
+}
+
+ManifoldManifold *manifold_transform(void *mem, ManifoldManifold *m, float x1,
+                                     float y1, float z1, float x2, float y2,
+                                     float z2, float x3, float y3, float z3,
+                                     float x4, float y4, float z4) {
+  auto mat = glm::mat4x3(x1, y1, z1, x2, y2, z2, x3, y3, z3, x4, y4, z4);
+  auto transformed = from_c(m)->Transform(mat);
+  return to_c(new (mem) Manifold(transformed));
+}
+
+ManifoldManifold *manifold_warp(void *mem, ManifoldManifold *m,
+                                ManifoldVec3 (*fun)(float, float, float)) {
+  std::function<void(glm::vec3 & v)> warp = [fun](glm::vec3 &v) {
+    v = from_c(fun(v.x, v.y, v.z));
+  };
+  auto warped = from_c(m)->Warp(warp);
+  return to_c(new (mem) Manifold(warped));
+}
+
+ManifoldMesh *manifold_level_set(void *mem, float (*sdf)(float, float, float),
+                                 ManifoldBox *bounds, float edge_length,
+                                 float level) {
+  return level_set(mem, sdf, bounds, edge_length, level, false);
+}
+
+ManifoldMesh *manifold_level_set_seq(void *mem,
```

Comment:
It'd be nice to not have two functions here; see my other comment above. But I can't tell if there's a way around it.

Reply:
I commented on the other one, but I think it is either this, or making the caller to provide a "boolean" (int) flag specifying that execution should be forced sequential. Unfortunately the factors that require avoiding parallel execution of the closure are external and not encoded into the type system as the distinction between static and dynamic lambdas are.

Reply:
Okay this makes sense now; can you add a comment on these bindings so people know they should really only link to one or the other depending on their language?

Reply:
Yeap I can do that.

Reply:
Added here: https://github.com/elalish/manifold/pull/292/commits/e25964bad67db377c3c193f787d020ccb62d1fb5